G4HepEm/G4HepEmRun/CMakeLists.txt

```diff
@@ -42,29 +42,22 @@ set(G4HEPEmRun_sources
 
 # See Issue #11 and discussion...
 set_source_files_properties(${G4HEPEmRun_impl_headers} PROPERTIES LANGUAGE CXX)
-if(BUILD_STATIC_LIBS)
-  add_library(g4HepEmRun STATIC ${G4HEPEmRun_sources} ${G4HEPEmRun_impl_headers})
-else()
-  add_library(g4HepEmRun SHARED ${G4HEPEmRun_sources} ${G4HEPEmRun_impl_headers})
-endif()
 
-set_target_properties(g4HepEmRun PROPERTIES COMPILE_FLAGS "-x c++ ${CMAKE_CXX_FLAGS}")
+g4hepem_add_library(g4HepEmRun
+  SOURCES ${G4HEPEmRun_sources} ${G4HEPEmRun_impl_headers}
+  HEADERS ${G4HEPEMRun_headers} ${G4HEPEmRun_impl_headers}
+  LINK g4HepEmData Geant4::G4clhep)
```

Comment:
Linking to `Geant4::G4clhep` won't work if the install of Geant4 being used was built with a system CLHEP. In that case, the targets are `CLHEP::CLHEP` and `CLHEP::CLHEPS`, either or both of which could be present. Likely needs the check for `Geant4::G4clhep` to be restored in the top level CMake script, plus a check for CLHEP targets. That'd be something in the top level CMake script like

```cmake
if(TARGET Geant4::G4clhep)
  set(G4HepEm_CLHEP_TARGET Geant4::G4clhep)
  set(G4HepEm_CLHEP_TARGET_static Geant4::G4clhep-static)
elseif(TARGET CLHEP::CLHEP)
  set(G4HepEm_CLHEP_TARGET CLHEP::CLHEP)
  set(G4HepEm_CLHEP_TARGET_static CLHEP::CLHEP) # Not a typo, Geant4 takes what it can get from CLHEP
endif()
```

then linking to these as appropriate.

_Aside: probably something to look at in Geant4's `Geant4Config.cmake` and targets to make `Geant4::G4clhep` an interface target to `CLHEP::CLHEP` if that was used._

Reply:
Thanks, that was the case I was missing. I'm going to keep it in `G4HepEm/G4HepEmRun/CMakeLists.txt` though